### PR TITLE
feat(account-lib): refactor mintAddress -> coinName STLX-11959

### PR DESCRIPTION
**Pull request by bitgoandrew**
Changing TokenTransfer class to store token name instead of mint address. This is so that in wallet platform, the token information can easily be obtained with coins.from(tokenName). The mint address requirements for certain splToken functions will be handled inside the TokenTransfer class to make futher development in other areas much simpler (ATA creation is based on token name as well)

### CODE REVIEW

**Review by joey-bai — APPROVED** [review]
LGTM.

**Review by mmcshinsky-bitgo — COMMENTED** [review]
To make sure, other files such as `ataInitializationBuilder.ts` won't need `mintAddress` removed or `tokenName` added for additional use cases?

File: modules/account-lib/test/resources/sol/sol.ts [hunk 1]
```diff
@@ -222,6 +222,7 @@ export const ATA_INIT_SIGNED_TX_WITH_MEMO =
   'AQB6vNNCKha4pL0Chx08W/7pqCRReR90+hA01EIgu5uycs+ZUeZgTrLcPSqaAcHDg4kPnjoAYZ/6+kCNGhFbywIBAAYIAGymKVqOJEQemBHH67uu8ISJV4rtwTejLrjw7VSeW6dNqdo+cuWQ+rQ8LcBm6GL6h1EGC0NXmyuOXCBESkJBmQwA0K/rhhTafxmroC1A8YxpJYX2UCDfztPV5fmpwMThAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAG3fbh12Whk9nL4UbO63msHLSF7V9bN5E6jPWFfv8AqQan1RcZLFxRIYzJTD1K8X9Y2u4Im6H9ROPb2YoAAAAAjJclj04kifG7PRApFI4NgwtaE5na/xCEBI572Nvp+FkFSlNamSkhBk0k6HFg2jh8fDW13bySu4HkH6hAQQVEjeMy2vkvq+zotj/3pEAF5f39mvoVh1a2HFqV+QSzuNCBAgYHAAEAAgMEBQAHABd0ZXN0IG1lbW8gcGxlYXNlIGlnbm9yZQ==';
 
 export const tokenTransfers = {
+  nameORCA: 'tsol:ORCA',
   mintORCA: '9cgpBeNZ2HnLda7NWaaU1i3NyTstk2c4zCMUcoAGsi9C',
```
[the remark this answers is not in the document]
Comment (bitgoandrew) [reply modules/account-lib/test/resources/sol/sol.ts:226]: It is obtained from coins.ts through my new way of handling the object now but still needed for other unit tests for instructionParamFactory. & solInstructionFactory!

File: modules/account-lib/src/coin/sol/instructionParamsFactory.ts [hunk 2]
```diff
@@ -23,9 +23,11 @@ import {
   StakingWithdraw,
   TokenTransfer,
 } from './iface';
-import { getInstructionType } from './utils';
+import { getInstructionType, getSolTokenFromAddress } from './utils';
```
[the remark this answers is not in the document]
Comment (bitgoandrew) [reply modules/account-lib/src/coin/sol/instructionParamsFactory.ts:26]: good catch, thank you

**Review by joey-bai — APPROVED** [review]

**Pull request merged**